File: src/spice2x/api/resources/dart/spiceapi/src/wrappers/control.dart
part of spiceapi;

Future<void> controlRaise(Connection con, String signal) {
  var req = Request("control", "raise");
  req.addParam(signal);
  return con.request(req);
}

Future<void> controlExit(Connection con, int code) {
  var req = Request("control", "exit");
  req.addParam(code);
  return con.request(req);
}

Future<void> controlRestart(Connection con) {
  var req = Request("control", "restart");
  return con.request(req);
}

Future<void> controlRefreshSession(Connection con) {
  var rnd = new Random();
  var req = Request("control", "session_refresh", id: rnd.nextInt(pow(2, 32)));
  return con.request(req).then((res) {
    con.changePass(res.getData()[0]);
  });
}

Future<void> controlShutdown(Connection con) {
  var req = Request("control", "shutdown");
  return con.request(req);
}

Future<void> controlReboot(Connection con) {
  var req = Request("control", "reboot");
  return con.request(req);
}

File: src/spice2x/api/resources/python/spiceapi/control.py
import random
from .connection import Connection
from .request import Request


def control_raise(con: Connection, signal: str):
    req = Request("control", "raise")
    req.add_param(signal)
    con.request(req)


def control_exit(con: Connection, code=None):
    req = Request("control", "exit")
    if code:
        req.add_param(code)
    try:
        con.request(req)
    except RuntimeError:
        pass  # we expect the connection to get killed


def control_restart(con: Connection):
    req = Request("control", "restart")
    try:
        con.request(req)
    except RuntimeError:
        pass  # we expect the connection to get killed


def control_session_refresh(con: Connection):
    res = con.request(Request("control", "session_refresh", req_id=random.randint(1, 2**64)))

    # apply new password
    password = res.get_data()[0]
    con.change_password(password)


def control_shutdown(con: Connection):
    req = Request("control", "shutdown")
    try:
        con.request(req)
    except RuntimeError:
        pass  # we expect the connection to get killed


def control_reboot(con: Connection):
    req = Request("control", "reboot")
    try:
        con.request(req)
    except RuntimeError:
        pass  # we expect the connection to get killed
